salt/pillar/__init__.py

```diff
@@ -378,6 +379,11 @@ def get_tops(self):
                         ]
             else:
                 for saltenv in self._get_envs():
+                    if self.opts.get('pillar_source_merging_strategy', None) == "none":
+                        if self.saltenv and saltenv != self.saltenv:
+                            continue
+                        if not self.saltenv and not saltenv == 'base':
```

Comment:
Typically we'd like to see these conditionals combined but here I think it enhances readability. Good call. :+1: